lib/use-cases/add-goal.js

```diff
@@ -0,0 +1,25 @@
+import Goal from '../domain/goal';
+
+export default class AddGoal {
+  constructor({ logger, planGateway }) {
+    this.logger = logger;
+    this.planGateway = planGateway;
+  }
+
+  async execute({ planId, goal }) {
+    const existingPlan = this.planGateway.get({ id: planId });
+
+    if (!existingPlan) {
+      this.logger.info('No plan found', { id: planId });
+      return null;
+    }
+
+    existingPlan.goal = new Goal({
```

Comment:
Would it override an existing goal if there was one?

Reply:
Yep. At this stage the only time a goal is added is on plan creation as far as i know

Reply:
👌